### ISSUE TITLE
Deciding and Documenting the Basics of the v5 Undefended "aaGun" Destruction Behaviour (2.0.16244)

### ISSUE THREAD

**Issue by Cernelius** [triplea-game/triplea#5363]
The undefended units destruction behaviour is problematic, and needs the following fixes and documentations:

1. Move this "total destruction" behaviour of defenceless units from start of each combat round to end of each combat round (immediately before the retreat prompt) and at the start of each battle.

2. Document (in pos2) exactly how the engine infers what units may be subjected to this behaviour and under which conditions it occurs.

---
Original post:

What is it that determines (in the program) the fact that the "aaGun" of "World War v5 Second Edition" are destroyed if alone?

I believe this behaviour should be defined in unit type's carateristics terms (for example, the unit must be exactly the same as a v5 aaGun, or it just needs having basic defence 0, or maybe current defence 0, or something else), and documented in pact of steel (first of all, I'm asking what is the current way this works), so that mapmakers may know what would cause such behaviour in units.

Even better would be an actual option for defining such a behaviour (destroyed if alone or only with infrastructures or other units with this option), then updating the v5 maps that need it (adding this option to the aaGun unit of that game).

I've a game in which there are att/def 0/0 immobile units (pure fodder, except for the fact that they turn into something else, when destroyed), and I've tested they are not autodestroyed if a single infantry like unit attacks them (I have to keep fighting till destroying them all).

Once this is done, I would need clarifications on the matter of destroyed damaged units turning into other units, in case they are removed under the v5 rule for unprotected aa guns, instead of being taken as casualties.

I hope this can be sorted out or, if I'm missing some explanations somewhere, please redirect me. Thanks.

**Comment by panther2** [comment]
Just adding here that the "autodestruction" of the defenseless transport or AA-gun in for example v5 (or ww2-1940-any) games is not a special characteristic of a unit. It is just caused by the fact that those units don't have a defense value in regular combat. "Autodestruction" is only a simplification for saving dice rolls against a unit that cannot defend rolling dice.
What makes it special is the fact that attacking units cannot retreat when facing only defenseless units.

**Comment by Cernelius** [comment]
@panther2 Based on what I recall about the official clarifications, what they say, to summarize, is that this rule is merely time-saving and not game changing at all, only saving you from having to keep rolling dice. However, this statement is contradicted by the fact that this rule, as you say, is actually more than that, as it makes you unable to retreat when facing only defenseless units (that also implies no strafing a stack of only such units).

Also based on this fact, if this behaviour for v5 aa guns (that I tested they work correctly) has been inferred in some ways, then I tend to think this is not a good way to apply it, or at least it should be very well documented how the engine decides if a unit type is subjected to this "autodestruction when alone" rule or not (let me know if I'm missing), especially for active mapmakers.

**Comment by panther2** [comment]
@Cernelius I don't see a contradiction here, as a simplification of rolling dice is different from retreat rules. 

The battle simply ends because there is nothing to retreat from.

I did not intend to negate your issue, I just wated to add that information.

**Comment by Cernelius** [comment]
@panther2 Actually, can I have some clarifications on the v5/Global aa gun intended behaviour (not talking about the transports here). Does the (obligatory) autodestruction rule apply only if I invade a territory with only aa guns and no other combat units, or also if I get to that situation during the course of the battle. In this case, when exactly does the autodestruction happen (in particular, can I retreat if, during a combat round, I've killed all defending combat units but one or more aa guns?).

Without the autodestruction rule, there are a number of things I could do, with defending units like the v5/Global aa guns, namely:
1- I could invade a territory having a stack of aa guns only in it, attacking all the aa guns, do one or more combat rounds, then retreat if not all aa guns have been destroyed (strafe).
2- I could invade a territory having aa guns and other combat units in it, and, if the defending player selects casualties so to have only aa guns left, I could retreat immediately after that happens, leaving only aa guns in the territory.
3- I could invade a territory having aa guns and other combat units in it, and, if the defending player selects casualties so to have only aa guns left, I could keep attacking for more combat rounds, possibly destroying some aa guns during the course of these combat rounds, then retreating, if in the territory there is still a positive number of aa guns left, leaving the surviving aa guns in the territory.

I've tested TripleA, and what the engine allows is the following:
1- No.
2- Yes.
3- No.

I believe this matter is not clarified by the referring official rulebook and FAQ, as only point 1 is obviously illegal. If I'm missing something, or the matter has been clarified anywhere else, please give me all the relevant links.

---

At any developers, what I'm mostly waiting for is someone telling me (or linking me to the documentation regarding) exactly how the program currently infers (I'm assuming it does) and applies this rule to units or, if I'm missing some attributes, let me know what I'm missing that causes the autodestruction behaviour for v5 aa guns. The next step would be pos2 documentation. This is the attachment for the mentioned game:
https://github.com/triplea-maps/world_war_ii_v5_1942/blob/master/map/games/WW2v5_1942_2nd.xml
```
    <attachment name="unitAttachment" attachTo="aaGun" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">

      <option name="movement" value="1"/>

      <option name="attack" value="0"/>

      <option name="defense" value="0"/>

      <option name="transportCost" value="3"/>

      <option name="canNotMoveDuringCombatMove" value="true"/>

      <option name="isAAforCombatOnly" value="true"/>

      <option name="maxAAattacks" value="3"/>

      <option name="isRocket" value="true"/>

    </attachment>
```
I'm not seeing any units options specifically for this "autodestruction" special rule, so I'm assuming the program is somewhat inferring this unit is subjected to that rule.

p.s.: As I said, this matter needs to be clarified considering that TripleA is not meant to support only the v5 standard games, but also v5 rules based games. For example, one could make a mod of v5 in which aa guns can fire before each combat round, as the only variation from the original game. In this case, if you end up with only attacking fighters and defending aa guns in the territory, autodestroying the aa guns would be a game-changing rule, as they would be not truly defenceless.

**Comment by panther2** [comment]
@Cernelius The only difference - when dealing with the situation in question - between the defending transport and the defending AAA is that transports must be chosen last as casualty.

AAA: "If a territory containing AAA units and no combat units is attacked, the AAA units are automatically destroyed."

Transport: "In a sea battle, if the defender has only transports remaining and the attacker still has units capable of attacking, the defending transports are all destroyed,..."

There is nothing to retreat from in case you either face only a AAA or only a Transport.

Retreat rules say:

>Step 6. Press Attack or Retreat
>Combat rounds (steps 2–5) continue unless one of the following two conditions occurs (in this order):
>
>Condition A Attacker and/or Defender Loses All Units
>"Once all units that can either fire at a valid target or retreat on one or both sides have been destroyed, the combat ends."
>
>Condition B Attacker retreats.
>...

If the defender has only AAA/transports remaining, he has lost all units that may either fire or retreat (they can do neither). One side has met that condition, so the combat is over. 
So condition B is not an option.

The decisive element is that as well transports as AAA have a combat value of 0 meaning they cannot fire during "Step 4. Defending Units Fire" part of the battle.

**Comment by Cernelius** [comment]
I suspected that the aa gun matter should have been exactly the same as the defenceless transport, except only for the fodder part. But I really have a hard time to understand why, in this case, this is not simply explained like that in the rulebook or in the FAQ. Can you link me anything official at all that states what you just explained. Anyways, I believe you, and I think this part you mentioned is what really clarifies it, though it is maybe clear enough, but not very clear, in my opinion:

>Once all units that can either fire at a valid target or retreat on one or both sides have been destroyed, the combat ends.

On the other hand, I don't think this part is very helpful:

>If a territory containing AAA units and no combat units is attacked, the AAA units are automatically destroyed

In my opinion this part is very unclear if by "attacked" it is meant that the territory was defended like that when initially attacked, or if also allowing for a condition happening in the course of the battle, while the territory is still under attack. Since it is talking about the territory, not the units, being attacked, it seems to hint to the former case.

On the other hand, the transport case is much clearer:
>In a sea battle, if the defender has only transports remaining and the attacker still has units capable of attacking, the defending transports are all destroyed

Why not just saying the same thing for the aa guns, as well?

Anyways, I suppose you can check and confirm that TripleA is currently bugged here too, then, as the aa guns alone are removed at the start of a combat round, leaving the attacker the option to retreat in the case number 2 I listed:
>I could invade a territory having aa guns and other combat units in it, and, if the defending player selects casualties so to have only aa guns left, I could retreat immediately after that happens, leaving only aa guns in the territory.

You can recreate this situation in "World War II v5 1942 Second Edition", editing, as I did.

So, practically (what a shock...), seems like we have some sort of magically implemented rule, that doesn't actually work correctly, and I'm not seeing where it is documented how a mapmaker is supposed to at least try to have it for a unit of its own custom game. This needs to be fixed and fully documented (unless I'm overlooking something, let me know).

**Comment by Cernelius** [comment]
@panther2 Since, as I said, we should consider customized games, what do you think should happen if I would make a mod of "World War II v5 1942 Second Edition" that is exactly like the original, except that infantry has offence=0 (but can still combat move, like the Classic transports) and, then, I attack with infantry only a territory defended by aa guns only (thus both all offenders and all defenders having 0 total power)?

This would be actually the case of a game I'm making, in which I have "psilos" at offence 0 and "wall" at defence 0.

**Comment by panther2** [comment]
@Cernelius 

> Can you link me anything official at all that states what you just explained.

Everything I explained is taken from the rulebook or an application of the rules on the situation.
So I am not sure what you are searching here. However I remember there have been questions and almost identical answering statements by Krieghund and/or me here and there addressing these topics. I will provide you with links when I find some time to search for those quotes.

> Anyways, I believe you, and I think this part you mentioned is what really clarifies it, though it is maybe clear enough, but not very clear, in my opinion:
>>Once all units that can either fire at a valid target or retreat on one or both sides have been destroyed, the combat ends.

Indeed, in the end it is only a check whether requirements to retreat are fulfilled or not.

> In my opinion this part is very unclear if by "attacked" it is meant that the territory was defended like that when initially attacked, or if also allowing for a condition happening in the course of the battle, while the territory is still under attack. 

A territory is attacked as well in the first round of combat as in later round of combats - but I understand your point.

> Why not just saying the same thing for the aa guns, as well?

Yes, just another (small) wording inconsistency.

> Anyways, I suppose you can check and confirm that TripleA is currently bugged here too, then, as the aa guns alone are removed at the start of a combat round, leaving the attacker the option to retreat in the case number 2 I listed: ...

Indeed, we have been dealing with retreat issues in the past and there is still an open issue addressing this:
https://github.com/triplea-game/triplea/issues/4688

> Since, as I said, we should consider customized games, what do you think should happen if I would make a mod of "World War II v5 1942 Second Edition" that is exactly like the original, except that infantry has offence=0 (but can still combat move, like the Classic transports) and, then, I attack with infantry only a territory defended by aa guns only (thus both all offenders and all defenders having 0 total power)?

This reminds me of the "Transport stalemate" issue:
https://github.com/triplea-game/triplea/issues/2367

Does that help?

**Comment by Cernelius** [comment]
So I believe we have clarified the main tasks here are:

1. Move this "total destruction" behaviour of defenceless units from start of each combat round to end of each combat round (immediately before the retreat prompt) and at the start of each battle.

2. Document exactly how the engine infers what units may be subjected to this behaviour under which conditions it occurs.

I'm updating the first post.

**Comment by Cernelius** [comment]
@panther2 What do you think about the fact that this behaviour allows to manage casualties selection to determine whether the turn player may be able to retreat or not?
For example, if I have 3 infantries and 2 aa guns, and get attacked while defending, receiving 3 or 4 hits, I can decide to take out all my infantries, so to deny the attacker the ability to retreat, that he would have if I would save at least 1 infantry.
Isn't that, at least theorically, a major departure from traditional rules, that until then have never allowed the defender to determine retreat possibilities on casualties selection?
I also wonder if the original rules-makers actually considered this matter at all.

Anyways, I really think it is seriously wrong to have inferred such a rule, if so. This behaviour should at least have a property, for determining whether having or not having it. I haven't found anything at all even to know about the existence of this behaviour, beside testing it or knowing the intended rules.

**Comment by panther2** [comment]
@Cernelius 
Here are my thoughts:

A&A rulesets through all the years have always been a continuous further development through editions. While the retreat rules have not changed, the new type of AAA has already been introduced with the transition from 1940 Europe/Pacific/Global 1st to 2nd edition, started with the so-called "Alpha3-ruleset" back in 2011. So v5 does not introduce a new behaviour but continues a behaviour that had been already implemented. So you most likely won't find much about these units in the v5 forum but rather buried deep somewhere in the Global forum categories. In the end v5 is nothing else than a feature-reduced Global1940.2, as you know. On A&A.org v5 never has been discussed as deeply as Global.

While v3 introduced the defenseless transport, 1940.alpha (1940.2) introduced the defenseless AAA that is no longer subject to capture but can be chosen as casualty. 

There has never been a "traditional rule" that the defender is not allowed to determine retreat possibilities. Retreat possibilities are determined by the retreat rules, as it has always been. And the retreat rules are checked after applying the results of dice and casualty selection of either side. 
(By the way when removing a defending surface warship instead of a defending submarine facing an attacking plane, the defender immediately ends the battle, too.) 
So nothing surprising here, IMHO.

The attacker simply has to take the AAA-charcteristics into account. The fact that Larry and Krieghund did not apply the "chosen last"-rule to AAA gives the defender some more freedom choosing casualties, compared to transports.

**Comment by Cernelius** [comment]
As another note, this is inconsistent with the similar unsupported rule intended for World War II Classic, albeit in this case it is for the offence, instead of for the defence.

While in the intended v3 ruleset if only defenceless defenders remains (either transports or aa guns), they are immediately automatically destroyed, without the option of retreating from them, in the intended Classic ruleset if only defenceless attackers remain (only transports), they are not immediately destroyed, but you are, instead, not only allowed, but even obliged to retreat, and are destroyed only if you cannot do so.

So, for the matter at hand, I suggest **keeping the current behaviour of automatical destruction at the start of the combat round as default** (as it is also correct for the first combat round only for v3+, to avoid unnecessary rolling)**, but expanding it as working both on the offence and the defence and adding a property that, if true, adds this behaviour also immediately before the retreat step, having the property "WW2V3" setting such property as true, as well.**

If this would be done, the v3+ behaviour of destroying defenceless defenders would be fully supported, and also the Classic behaviour of destroying defenceless offenders that don't retreat would be mostly supported, aside only from the item of obliging the offender to retreat, if it can (this could be fixed, as well, by not having the option of continue the battle if all attackers are defenceless and you have one or more retreat options).

Having such a property would be also good so that the mapmakers may choose, for custom games, if they want to give the option of retreating either in face or defenceless opponents or when being left defenceless as the attacker, or not (I personally prefer the current behaviour over the v3 ones, as I feel it is very strange to just sacrifice your units so to deny retreat to the enemy, and definitely realistically absurd).

And, of course, the other item is deciding and documenting exactly when all units of one side are to be considered defenceless (for example, should they be considered as such if they are currently all defenceless due to negative supports?).

@Panther Do you agree with this suggestion on how to fix this problem?

**Comment by Cernelius** [comment]
This change would, of course, also change be behaviour of all games having attack 0 units that are currently not covered by any autodestruction rule. For example, in the "Lord of the Rings: Middle Earth" having only hobbits at the start of each combat round of an ongoing battle will immediately cause all of them to be killed (in practice, this will remove the option of moving by strafing a territory with hobbits only). I think this would be consistent the most with the rules of the referring rulesets, as in those rulesets only offending transports were affected, but I see this more like a general rule that should apply to whatever attack 0 units (that in the original rulesets were the transports only) (so, I believe you should not be allowed to strafe with hobbits only just like you must not be allowed to strafe with transports only).

**Comment by panther2** [comment]
@Cernelius 

> While, in the intended v3 ruleset, if only defenceless defenders remain (either transports or aa guns), they are immediately automatically destroyed, without the option of retreating from them,

In v3 this applies to the defenseless transport, only. The AAA is subject to capture, here.

>Do you agree with this suggestion on how to fix this problem?
As long as attacking units with an attack value of "0" have legally entered the battle - and then have been given the option to retreat after the first round of combat, this sounds reasonable.

**Comment by Cernelius** [comment]
>As long as attacking units with an attack value of "0" have legally entered the battle

Well, it is illegal sending only transports in a battle, also in Classic (which I would read as being illegal sending only attack value 0 units in a battle), and this restriction is not enforced by TripleA, but, as said, here at this issue I'm not suggesting implementing that too, as well as the fact that you are obliged to retreat, if you can, if left in the same condition.

Anyways, I guess I could add these item too, updating the first post.

**Comment by simon33-2** [comment]
I think this issue is a duplicate of #4688.

I suggest we have just one issue to track this. I actually agree with @Cernelius that the rule is silly and you shouldn't be allowed to deliberately sacrifice your units to prevent a strafe.

I also support his idea that this should be controlled by properties. Or perhaps they already are. If transports are taken as hits last, there is still the issue that subs can submerge (which is illegal) but there is also an issue for this (#4821). Perhaps there should be a "Allow retreat against defenseless defenders" property. Would that work for everyone?

**Comment by Cernelius** [comment]
@simon33-2 This issue was originally focused on knowing what is it that determines (in the program) the fact that the "aaGun" of "World War v5 Second Edition" are destroyed if alone.

This problem exists no matter if the autodestruction happens at start of any combat rounds (as the program works and as I prefer it) or anytime (as it should work).

The problem with that other issue is that leaves a number of problems unchallenged (sure, if I made that issue, I could have just edited that one).

**Comment by simon33-2** [comment]
So what are you looking to get from this issue? As far as I am aware, it's a non optional feature, which sometimes doesn't quite work properly.

**Comment by Cernelius** [comment]
>So what are you looking to get from this issue? As far as I am aware, it's a non optional feature, which sometimes doesn't quite work properly.

?

It's point 2 written above, and most of what I said in the original post.

>Document (in pos2) exactly how the engine infers what units may be subjected to this behaviour and under which conditions it occurs.

Look, if TripleA would be only about supporting Global, etc., the other issue would be enough, but how about if I want to make some custom game with some units having defence 0 (like the trenches of NML) or having attack 0 (like the hobbits of LoTR). If and how that is going to be subjected to the v5 autodestruction rule (mentioning v5, because before that it was for sea units only), should be definite, possibly definitive, and fully documented (also keep in mind that I may have v5 AA Guns except that they fire at each combat round; so here maybe they should be autodestroyed only if facing non-air units only, or maybe not? (I believe it can go either ways, but it has to go one way, and it should be documented)).

Basically what I believe, and said, it is that either you make a custom option, to assign specifically to units, for this autoremoval behaviour, and it should be exactly and fully documented when it happens, talking from a TripleA, not the original rulebooks, standpoint.

**Comment by Cernelius** [comment]
>I actually agree with @Cernelius that the rule is silly and you shouldn't be allowed to deliberately sacrifice your units to prevent a strafe.

Yep. This is why I definitely believe it would be nice keeping the current behaviour (always being able to retreat in face of defenceless units only, yet the combat ending at start of any round, if in such a condition) for custom games. However, for the standard games, it is what it is, thus that is a tactic you should be able to perform, however silly.

**Comment by Cernelius** [comment]
I've tested the current defenceless autodestruction behaviour applies all the same both to the "aaGun" of "World War II Global 1940 2nd Edition" and the "trench" of "Domination 1914 No Man's Land".

These are the attachments:
```
    <attachment name="unitAttachment" attachTo="aaGun" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
      <option name="isAAforCombatOnly" value="true"/>
      <option name="canNotMoveDuringCombatMove" value="true"/>
      <option name="movement" value="1"/>
      <option name="transportCost" value="3"/>
      <option name="canBeGivenByTerritoryTo" value="British"/>
      <option name="maxAAattacks" value="3"/>
    </attachment>
```
```
    <attachment name="unitAttachment" attachTo="trench" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
      <option name="defense" value="0"/>
      <option name="hitPoints" value="2"/>
      <option name="isSuicide" value="true"/>
      <option name="isConstruction" value="true"/>
      <option name="constructionType" value="bunker"/>
      <option name="constructionsPerTerrPerTypePerTurn" value="3"/>
      <option name="maxConstructionsPerTypePerTerr" value="1000"/>
    </attachment>
```
However, in an unreleased game of mine, I have this unit:
```
                <attachment name="unitAttachment" attachTo="wall" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
                         <option name="movement" value="0"/>
                         <option name="canNotMoveDuringCombatMove" value="true"/>
                         <option name="attack" value="0"/>
                         <option name="defense" value="0"/>
                         <option name="hitPoints" value="1"/>
                         <option name="isConstruction" value="true"/>
                         <option name="constructionType" value="fortification"/>
                         <option name="constructionsPerTerrPerTypePerTurn" value="1000"/>
                         <option name="maxConstructionsPerTypePerTerr" value="1000"/>
                         <option name="requiresUnits" value="metropolis"/>
                </attachment>
```
And the autodestruction is, instead, not happening.

I also tried to remove the `canNotMoveDuringCombatMove` part, but nothing changed.

Then I've added the properties:
```
    <property name="WW2V3" value="true" editable="false">
      <boolean/>
    </property>
    <property name="WW2V3 Tech Model" value="true" editable="false">
      <boolean/>
    </property>
```
But, it still isn't working.

So, I've removed the `requiresUnits` option, but that didn't do it, either.

Then, I just resolved pasting the code for the "aaGun", but removing the `canBeGivenByTerritoryTo` and calling it "wall", thus having:
```
    <attachment name="unitAttachment" attachTo="wall" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
      <option name="isAAforCombatOnly" value="true"/>
      <option name="canNotMoveDuringCombatMove" value="true"/>
      <option name="movement" value="1"/>
      <option name="transportCost" value="3"/>
      <option name="maxAAattacks" value="3"/>
    </attachment>
```
And it is still not working.

So, this is an example why behaviours, that I assume the program infers somehow, should be exactly documented, in pos2.

**Anyways, if anyone around have any clue about what I should do to have that behaviour for a unit like the one I mentioned, just let me know here.**
